### DISCUSSION TITLE
Integration Test Failures

### DISCUSSION THREAD

**Discussion by seanvaleo** [appview-team/appview#22] [General]
from @iapaddler :
A few thoughts on integ tests:

    remove Go 2-14
    real issues to debug: awsnuke, cli, console, gogen, Go 11-20, http, postgres, splunk, terraform, tls, transport, alpine
    require newer version of OpenSSL, kafka
    build issues that need to be explored; java7, opentelemetry, syscalls

there are some common issues in 2 above:

    all tests using apache fail DNS for some server
    all Go executables appear to have a common segfault issue

**Comment by iapaddler** [comment]
Alpine: this is probably prio 1
Tests were updated to be based on Alpine:latest. It was based on alpine:3.15.
Tests for Python, Ruby, Rust started to fail.
The issue was that the GOT detail is different for some reason. 
The length of any given entry returns 1 function. This, of course, inhibits all functions from being interposed.
Need to determine that change in structure that has occurred so that the GOT can be parsed effectively. 

Tests affected by alpine:latest
Changed attach-musl and payload-musl to use alpine:3.15 and not alpine:latest.
These need to be put back when the GOT issue with alpine:latest is resolved.

Removing Go 2-15 from integ tests. A lot of the releases have been removed.
Latest supported Go version has been moved to 19.
Go 16, 17, 18 are built and run unsupported tests.

Removed influxdb tests from Go integ. The pre-built influx execs segfault without any lib. Should use influx packages and update tests accordingly.

Removed opentelemtry test. There are build errors with opentelemetry.  A build option that we specify called `WITH_OLTP` is now deprecated. Tried building without that option and with  latest version as opposed to the version specified. Also, we build this on alpine. According to the github site for this supported platforms do not include alpine. Seems like the next step is to use a supported platform such as ubuntu 22.

terraform fails with a segfault in the plan command. Need to debug this. Disabling for now.

splunk fails in unviewed mode, without appview. First place to look might be the older version being used. Disabling for now.

Rust on Alpine is failing. Removed from the test_ssl.sh test scripts.
The test downloads sample code. The Rust build downloads its' own libs, including libssl. The version of libssl that is used is newer, at least different. It uses SSL_write_ex and SSL_read_ex as opposed to SSL_write/read. Not a big difference. However, we will need to patch different funcs and provide modified interposed functions for the *_ex variants. Here's the diff in the SSL functions:
int SSL_write_ex(SSL *s, const void *buf, size_t num, size_t *written)
int SSL_write(SSL *ssl, const void *buf, int num)
int SSL_read_ex(SSL *ssl, void *buf, size_t num, size_t *readbytes)
int SSL_read(SSL *ssl, void *buf, int num)
